src/map.cpp

```diff
@@ -4298,6 +4298,18 @@ item map::water_from( const tripoint &p )
         }
         return ret;
     }
+    if( terrain_id == t_water_moving_sh ) {
```

Comment:
I think you could probably check the `DEEP_WATER` and `CURRENT` flags instead of checking `t_water_sh`, `t_water_dp`, `t_water_moving_sh`, `t_water_moving_dp` explicitly.

Reply:
I think that did it...